pkg/agent/controller/cleanup.go

```diff
@@ -0,0 +1,76 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/syncer/broker"
+	lhconstants "github.com/submariner-io/lighthouse/pkg/constants"
+	discovery "k8s.io/api/discovery/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog"
+)
+
+func (a *Controller) Cleanup() error {
+	err := a.cleanupResources(a.serviceImportClient.Namespace(metav1.NamespaceAll), a.serviceImportSyncer, &metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	return a.cleanupResources(a.endpointSliceClient.Namespace(metav1.NamespaceAll), a.endpointSliceSyncer, &metav1.ListOptions{
```

Comment:
If broker is on data cluster, this will end up deleting everything on broker. Does uninstall run when a cluster is unjoined? We should filter out broker namespace and in broker namespace filter out anything that is not local.